File: web2-project/src/app/model/incident.ts
import { Call } from "./call";
import { Device } from "./device";
import { Team } from "./team/team.model";

export enum IncidentType
{
    Planiran,
    Neplaniran
}


//nezavrsena klasa
export class Incident {
    public id:number;
    public userID:string;
    //public crewID:number;
    public tip:IncidentType;
    public prioritet:number = 0;
    public potvrdjen:boolean;
    public status:string;
    public eta:Date;
    public ata:Date;
    public vremeIncidenta:Date;
    public etr:Date;
    public nivoNapona:number;
    public pvr:Date; //planirano vreme rada
    public Devices:Array<Device>;
    public Calls:Array<Call>;
    public Team:Team;
    public uzrok:string;
    public poduzrok:string;
    public konstrukcija:string;
    public materijal:string;


    /*
    constructor(tip:IncidentType,status:string,eta:string,ata:string,vremeincidenta:string,etr:string,nivonapona:number,pvr:string)
    {
       
        this.tip = tip;
        //this.Prioritet = 0;
        //this.Potvrdjen = false;
        this.status = status;
        this.eta = eta;
        this.ata = ata;
        this.vremeIncidenta = vremeincidenta;
        this.etr = etr;
        this.nivoNapona = nivonapona;
        this.pvr = pvr;
        this.Devices = new  Array<Device>();
        this.Calls = new Array<Call>();
        //this.Uzrok = "";
        //this.Poduzrok = "";
        //this.Konstrukcija = "";
        //this.Materijal = "";
        
        
    }*/
}

File: Web2BackEnd/Web2BackEnd/Models/Incident.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Web2BackEnd.Models
{
    public class Incident:GenericModel
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }
        [Required]
        public IncidentType Tip { get; set; }
        
        public int? Prioritet { get; set; }
    
        public bool? Potvrdjen { get; set; }
        [Required]
        public string Status { get; set; }
        [Required]
        public string ETA { get; set; }
        [Required]
        public string ATA { get; set; }
        [Required]
        public string VremeIncidenta { get; set; }
        [Required]
        public string ETR { get; set; }
        [Required]
        public float NivoNapona { get; set; }
        [Required]
        public string? PVR { get; set; }
        
        public string Uzrok { get; set; }
       
        public string Poduzrok { get; set; }
      
        public string Konstrukcija { get; set; }
        
        public string Materijal { get; set; }

        //public ICollection<Device> Devices { get; set; }
        //public ICollection<Call> Calls { get; set; }
        //public Team team { get; set; }
        

        

    }

    public enum IncidentType
    {
        PLANIRAN,
        NEPLANIRAN
    }
}
